ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/manager/task/AbstractEthTask.java

```diff
@@ -29,19 +31,27 @@
 
 public abstract class AbstractEthTask<T> implements EthTask<T> {
 
+  static volatile LabelledMetric<OperationTimer> ethTasksTimer = null;
+
   protected double taskTimeInSec = -1.0D;
-  protected final LabelledMetric<OperationTimer> ethTasksTimer;
   protected final OperationTimer taskTimer;
   protected final AtomicReference<CompletableFuture<T>> result = new AtomicReference<>();
   protected volatile Collection<CompletableFuture<?>> subTaskFutures =
       new ConcurrentLinkedDeque<>();
 
-  /** @param ethTasksTimer The metrics timer to use to time the duration of the task. */
-  protected AbstractEthTask(final LabelledMetric<OperationTimer> ethTasksTimer) {
-    this.ethTasksTimer = ethTasksTimer;
+  protected AbstractEthTask(final MetricsSystem metricsSystem) {
+    setupTaskTimer(metricsSystem);
     taskTimer = ethTasksTimer.labels(getClass().getSimpleName());
   }
 
+  private synchronized void setupTaskTimer(final MetricsSystem metricsSystem) {
+    if (ethTasksTimer == null) {
+      ethTasksTimer =
+          metricsSystem.createLabelledTimer(
+              MetricCategory.SYNCHRONIZER, "task", "Internal processing tasks", "taskName");
```

Comment:
This has a race condition where separate instances could both wind up creating the timer at the same time with one of them likely failing because of the duplicate name check.  The `synchronized` only restricts it to a single thread per instance because the method isn't static.